lnst/Controller/RunSummaryFormatters/JsonRunSummaryFormatter.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional
+import logging
+import json
+
+from lnst.Controller.Recipe import RecipeRun
+from lnst.Controller.RecipeResults import DeviceMethodCallResult, JobResult, JobStartResult, MeasurementResult, Result
+from .RunSummaryFormatter import RunSummaryFormatter
+
+
+class JsonRunSummaryFormatter(RunSummaryFormatter):
+    def __init__(self, pretty: bool = False):
+        super().__init__()
+        self.pretty = pretty
+
+    def format_run(self, run: RecipeRun) -> str:
+        measurement_results = [
+            transformed
+            for result in run.results
+            if (transformed := self._transform_result(result)) is not None
+        ]
+
+        return json.dumps(
+            measurement_results,
+            indent=4 if self.pretty else None,
+        )
+
+    def _transform_result(self, result: Result) -> Optional[dict]:
```

Comment:
imo we should also include the `result.result` property: https://github.com/LNST-project/lnst/blob/master/lnst/Controller/RecipeResults.py#L74 for the individual types

Reply:
Right. Is the string representation (`"PASS"`, `"FAIL"`) enough? If yes, then this is done.